Wangscape/Reseedable.cpp

```diff
@@ -0,0 +1,285 @@
+#include "Reseedable.h"
+#include "MakeReseedable.h"
+#include "ModuleGroup.h"
+#include "Exp.h"
+
+double Reseedable::getValue(double x, double y, double z)
+{
+    return module->GetValue(x, y, z);
+}
+
+Reseedable Reseedable::abs()
+{
+    std::shared_ptr<noise::module::Abs> abs_p = std::make_shared<noise::module::Abs>();
```

Comment:
Yes, `auto` would be appropriate here.

`abs_p` can't be `const` because:
* `abs_p->SetSourceModule(0, *module);` is a non-const function call.
* `abs_p` will be stored in a `Reseedable`, and `Reseedable` stores a non-const `ModulePtr` because the `SetSeed` function may alter that `ModulePtr`.

In this case, `abs_p` does not have a seed to set, so the `SetSeed` function will be a no-op. But the current version of the `Reseedable`/`ModuleGroup` system can't accommodate a const `ModulePtr`.

However, this system is quite fragile as it stands. `Reseedable` and `ModuleGroup` data members are public and type information is not included. This means one could write code that alters the network of noise modules after construction, but it would need type information stored externally (or from RTTI), and would almost certainly have errors.

My first instinct would be to refactor the code with as much as possible `private`/`protected`/`const`. However, when implementing #17, that would probably mean that the whole network needs to be rebuilt when altered. That might not be a bad thing.